deepdiff/search.py

```diff
@@ -205,7 +205,7 @@ def __search_dict(self,
 
             str_item = str(item)
             if (self.match_string and str_item == new_parent_cased) or\
-               (not self.match_string and str_item in new_parent_cased):
+               (not self.match_string and re.search(str_item, new_parent_cased)):
```

Comment:
Since the regex will be running on objects repeatedly, it is going to be better to compile it beforehand and use it here. For example you could take a look at how `convert_item_or_items_into_compiled_regexes_else_none` is used for exactly the same purpose when the user passes `exclude_regex_paths`:

https://github.com/seperman/deepdiff/blob/5b66b96f1fa63f8ecc3b267e4a65c552df7d0190/deepdiff/diff.py#L177

https://github.com/seperman/deepdiff/blob/5b66b96f1fa63f8ecc3b267e4a65c552df7d0190/deepdiff/diff.py#L389